app/src/main/java/com/nervousfish/nervousfish/modules/database/GsonDatabaseAdapter.java

```diff
@@ -1,31 +1,184 @@
 package com.nervousfish.nervousfish.modules.database;
 
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.reflect.TypeToken;
+import com.nervousfish.nervousfish.data_objects.Contact;
+import com.nervousfish.nervousfish.data_objects.IKey;
+import com.nervousfish.nervousfish.modules.constants.IConstants;
 import com.nervousfish.nervousfish.service_locator.IServiceLocator;
 import com.nervousfish.nervousfish.service_locator.IServiceLocatorCreator;
 import com.nervousfish.nervousfish.service_locator.ModuleWrapper;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.lang.reflect.Type;
+import java.util.ArrayList;
+import java.util.List;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+
 /**
- * An adapter to the Gson database library
+ * An adapter to the GSON database library
  */
 public final class GsonDatabaseAdapter implements IDatabase {
 
+    private final static Type TYPE_CONTACT_LIST = new TypeToken<ArrayList<Contact>>(){}.getType();
+    private final IConstants constants;
+    private final GsonBuilder gsonBuilder = new GsonBuilder()
+            .registerTypeHierarchyAdapter(IKey.class, new KeyAdapter());
+    private final Gson gsonParser = this.gsonBuilder.create();
+
     /**
      * Prevents construction from outside the class.
+     *
      * @param serviceLocatorCreator The object responsible for creating the service locator
      */
-    @SuppressWarnings("PMD.UnusedFormalParameter")
     private GsonDatabaseAdapter(final IServiceLocatorCreator serviceLocatorCreator) {
-        // final IServiceLocator serviceLocator = serviceLocatorCreator.getServiceLocator();
+        final IServiceLocator serviceLocator = serviceLocatorCreator.getServiceLocator();
+        this.constants = serviceLocator.getConstants();
```

Comment:
It will probably be merged before this one, so I think it'll be OK. I'll update the PR description for clarity tho 🙂